### PR TITLE
Check if value in action logs message is null

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Randomly when I run my DPMedia system tests, a deprecated warning pops up in the action logs module in the back end dashboard. I couldn't identify which core extension is writing a null value as it happens with DPMedia which has no action logs plugin, so it must be a core extension.

Nevertheless this should be caught as it is very much possible to write null values into the action logs table.

Review should be enough.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_actionlogs/src/Helper/ActionlogsHelper.php [hunk 1]
```diff
@@ -200,7 +200,7 @@ public static function getHumanReadableLogMessage($log, $generateLinks = true)
 
         foreach ($messageData as $key => $value) {
             // Escape any markup in the values to prevent XSS attacks
-            $value = htmlspecialchars($value, ENT_QUOTES, 'UTF-8');
+            $value = $value ? htmlspecialchars($value, ENT_QUOTES, 'UTF-8') : '';
```
Comment (SharkyKZ) [inline administrator/components/com_actionlogs/src/Helper/ActionlogsHelper.php]: Use null or string check.

**Comment by sandewt** [comment]
> I couldn't identify which core extension is writing a null value as it happens with DPMedia which has no action logs plugin, so it must be a core extension.

Maybe the following code will give you a little more information?

```
$value = trim($value);

if (empty($value))
{
	unset($messageData[$key]);
	continue;
}

$value = htmlspecialchars($value, ENT_QUOTES, 'UTF-8');
```

screen shot 2022-12-15 at 08 43 46<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39421">issues.joomla.org/tracker/joomla-cms/39421</a>.</sub>

File: administrator/components/com_actionlogs/src/Helper/ActionlogsHelper.php (hunk 1 above)
Comment (laoneo) [reply administrator/components/com_actionlogs/src/Helper/ActionlogsHelper.php]: Make a suggestion

**Comment by laoneo** [comment]
@sandewt why should it? Where is this code located?

**Comment by sandewt** [comment]
>why should it?

Line 203: `$value = $value ? htmlspecialchars($value, ENT_QUOTES, 'UTF-8') : '';` 

In case of `$value = ''`, you will see less information, compared to the image above.

I don't know what exactly happens with DPMedia. I hope you can follow my reasoning.
issue_actionlog-2

>  Where is this code located?

Line 203 ...

**Comment by laoneo** [comment]
It is not about the code change. All I wanted to say that it happens randomly during the tests which I can't pause and identify which extension is causing the issue.

**Comment by sandewt** [comment]
> It is not about the code change. All I wanted to say that it happens randomly during the tests which I can't pause and identify which extension is causing the issue.

I understand, I wanted to offer you an alternative that provides just a little more information.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39421">issues.joomla.org/tracker/joomla-cms/39421</a>.</sub>

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on abf9e136d04356b88b8c5ee8e41f74f4f9d3eea3<br /><br />Code review !<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39421">issues.joomla.org/tracker/joomla-cms/39421</a>.</sub>

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on abf9e136d04356b88b8c5ee8e41f74f4f9d3eea3<br /><br />Code viewed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39421">issues.joomla.org/tracker/joomla-cms/39421</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39421">issues.joomla.org/tracker/joomla-cms/39421</a>.</sub>

**Comment by roland-d** [comment]
Thank you

**Pull request merged**